### PR TITLE
[build.webkit.org] Support identifiers in popover

### PR DESCRIPTION
**Pull request by JonWBedard**
#### 0975a44109764e822959c8935dc753764e637b90
<pre>
[build.webkit.org] Support identifiers in popover
<a href="https://bugs.webkit.org/show_bug.cgi?id=239773">https://bugs.webkit.org/show_bug.cgi?id=239773</a>
&lt;rdar://problem/92339575 &gt;

Rubber-stamped by Aakash Jain.

* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/Commits.js:
(Commits): Keep track of all fetched commits.
(Commits.prototype._update): Save fetched commits.
(Commits.prototype.lastNIdentifiers): Return a list of the last n identifiers
on a branch.
(Commits.prototype.fetch): Load information for last N commits.
(Commits.prototype.urlFor): Return a url for a commit ref.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/Main.js: Remove
trac auto-updater.
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/QueueView.js:
(QueueView.prototype._appendPendingRevisionCount): After we determine the
number of commits behind, fetch that many commits.
(QueueView.prototype._popoverLinesForCommitRange): Support commits with identifiers.
(QueueView.prototype._presentPopoverForPendingCommits):
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/tests/resources/MockCommits.js:
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/tests/resources/tests.js:

Canonical link: <a href="https://commits.webkit.org/250011@main">https://commits.webkit.org/250011@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@293475">https://svn.webkit.org/repository/webkit/trunk@293475</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

### CODE REVIEW

File: Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/QueueView.js [hunk 1]
```diff
@@ -199,20 +196,36 @@ QueueView.prototype = {
...
+            var lines = this._popoverLinesForCommitRange(commits, branch, commitList);
             var length = lines.length;
             if (length && shouldAddDivider)
                 this._addDividerToPopover(content);
             for (var j = 0; j < length; ++j)
                 content.appendChild(lines[j]);
             shouldAddDivider = shouldAddDivider || length > 0;
+
+            console.log(commitList);
+            /*if (!latestProductiveRevisionNumber || !trac.latestRecordedRevisionNumber)
```
Comment (aj062) [inline Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/QueueView.js]: is this code left unintentionally?

**Review by aj062 — APPROVED** [review]
rs=me

File: Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/QueueView.js (hunk 1 above)
Comment (JonWBedard) [reply Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/QueueView.js]: No, forgot to remove it.

**Comment by JonWBedard** [comment]
Will need to update https://github.com/WebKit/WebKit/pull/349 for this

File: Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/QueueView.js [hunk 2]
```diff
@@ -147,41 +149,36 @@ QueueView.prototype = {
...
-            titleElement.innerHTML = commit.title.innerHTML;
+            titleElement.innerHTML = commit.message ? commit.message.split('\n')[0] : '';
             result.appendChild(titleElement);
 
             return result;
         }
 
-        console.assert(trac.indexOfRevision(trac.oldestRecordedRevisionNumber) <= trac.indexOfRevision(firstRevisionNumber));
-
-        // FIXME: To be 100% correct, we should also filter out changes that are ignored by
```
Comment (aproskuryakov) [inline Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/QueueView.js]: I think that this is still an accurate comment.

Comment (JonWBedard) [reply Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/QueueView.js]: It is still an accurate comment, but I don't see any path to a fix. We would need to duplicate Buildbot's filter logic or have commits.webkit.org own the concept of filtering. I think both are bad ideas, and the benefit is pretty dubious in the first place

Comment (aproskuryakov) [reply Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/QueueView.js]: It wouldn't be too hard to fix actually, just need to post the rules as a JSON somewhere, factoring out of Python code.

Comment (JonWBedard) [reply Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/QueueView.js]: The rules are dependent on the files that changed, that information is possible to get, but isn't performant to get in the general case

**Pull request merged**